// make a closure for a function that is a retry of calling a callback function and you needed to retry for up to 10 times in case of error.
